### ISSUE TITLE
Don't stop the run if one of the benchmarks failed.

### ISSUE THREAD

**Issue by mateusz-blaszkowski** [GoogleCloudPlatform/PerfKitBenchmarker#573]
When you run multiple benchmarks (i.e. `--benchmarks=mongodb_ycsb,cassandra_ycsb`) and the first one fails, all the subsequent benchmarks will not be run. This is particularly unwanted when we want to run tests at night and it turns out that most of the benchmarks were not executed at all because of the first one failing.

**Comment by skschneider** [comment]
I believe this is also a current discrepancy between the single-threaded and parallel behavior. If you set --parallelism=2, it should run all benchmarks and then raise a single exception at the end that reports failures raised during any of the runs. But with the current default of --parallelism=1, you get the fail-and-stop behavior you mentioned.

Unless there's any reason to preserve the old behavior, I think continuing after a failure should be the new default.

**Comment by voellm** [comment]
I'm ok with continuing but we need to make sure the failed runs are
reported at the end clearly.  Not sure that happened now.

We should also have a flag like --continue_runs_on_failure and default to
true


On Wed, Oct 21, 2015 at 11:02 AM, skschneider <notifications@github.com>
wrote:

> I believe this is also a current discrepancy between the single-threaded
> and parallel behavior. If you set --parallelism=2, it should run all
> benchmarks and then raise a single exception at the end that reports
> failures raised during any of the runs. But with the current default of
> --parallelism=1, you get the fail-and-stop behavior you mentioned.
>
> Unless there's any reason to preserve the old behavior, I think continuing
> after a failure should be the new default.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/573#issuecomment-149979050>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/
Benchmark like a pro... PerfKitBenchmarker
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>

**Comment by mateusz-blaszkowski** [comment]
Having something like failure rate at the end of the run would be great. If I run dozens of benchmarks (either sequentially or in parallel) I would like to know how many benchmarks failed (and what were the reasons of failures).

**Comment by voellm** [comment]
Skyler - assigning to you.  Feel free to ask Noah to do it if you dont have time.

**Comment by cmccoy** [comment]
In case it's useful to anyone until a fix is available: I usually use [`parallel`](http://www.gnu.org/software/parallel/) to manage running multiple benchmarks; it can optionally output a log file with the status of each invocation (`--joblog`).

**Comment by skschneider** [comment]
Working on this now. Solution will be related to #529, and it may not be in the current release but will definitely be in by the release after that.

**Comment by mateusz-blaszkowski** [comment]
@skschneider - are you going to present some kind of a failure rate (with the list of benchmarks that failed) at the end of a run?

**Comment by skschneider** [comment]
Yes, that list & percentage are included in #597. They will be printed after the list of samples at the end of a run.

**Comment by skschneider** [comment]
Plans have changed to remove the --parallelism flag (#616). I have opened a new pull request (#614) that specifically addresses the current issue.

**Comment by skschneider** [comment]
This is fixed by #614 with a printed summary added in #620.

**Comment by mateusz-blaszkowski** [comment]
Looks great and the table in the end very clearly outlines the final status. Thank you for your work!
